### PR TITLE
enhance colorPaicker ally Control

### LINKED ISSUE #2924: Enhance the accessibility of `ColorPicker`component
### Description

This issue aims to address accessibility deficiencies in our slider component in accordance with W3C guidelines. The component currently does not meet the accessibility standards outlined in the W3C documentation.

The goal is to ensure the component is accessible to all users, including those using assistive technologies. All unchecked items in the checklist below should be addressed.

- `role`
	- [x] The `input` tag in the `ColorPicker` component should have the `combobox` role.
- `aria-controls`
	- [x] An element with the `combobox` role must have the `aria-controls` attribute containing the ID of the corresponding popup element.

### Are you willing to participate in fixing this issue and create a pull request with the fix?

None

### Additional Information
The following issue may be referenced:
https://github.com/yamada-ui/yamada-ui/issues/1731

### PR DESCRIPTION
**Pull request by taku10101**
Closes #2924 

## Description

add aria-controls and role

## Is this a breaking change (Yes/No):
no

### CODE REVIEW

File: .changeset/cool-cycles-build.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@yamada-ui/color-picker": minor
```
Comment (kaehehehe) [inline .changeset/cool-cycles-build.md:2]:
```suggestion
"@yamada-ui/color-picker": patch
```

**Review by kaehehehe — CHANGES_REQUESTED** [review]

**Comment by kaehehehe** [comment]
Two tests have failed. Please check.　@taku10101 

스크린샷 2024-09-26 17 26 24
d. Please check.

**Comment by kaehehehe** [comment]
The tests are failing right now, so could you pull the main branch and push it again? @taku10101

**Review by kaehehehe — APPROVED** [review]
LGTM!  💯 💯 💯

**Review by kaehehehe — CHANGES_REQUESTED** [review]

File: packages/components/color-picker/src/color-picker.tsx [hunk 2]
```diff
@@ -102,6 +102,13 @@ export const ColorPicker = forwardRef<ColorPickerProps, "input">(
       withSwatch,
       ...props,
     })
+    const contentRef = useRef<HTMLDivElement>(null)
+
+    useEffect(() => {
+      if (contentRef.current) {
+        contentRef.current.removeAttribute("role")
+      }
+    }, [])
```
[the remark this answers is not in the document]
Comment (taku10101) [reply packages/components/color-picker/src/color-picker.tsx:111]: @kaehehehe 
If you turn it off, you will get this Incomplete
Is it safe to ignore this?

https://dequeuniversity.com/rules/axe/4.10/aria-allowed-role?application=axeAPI
スクリーンショット 2024-09-29 12 43 45

[the remark this answers is not in the document]
Comment (kaehehehe) [reply packages/components/color-picker/src/color-picker.tsx:111]: It's not needed, so just delete it.  @taku10101

**Review by kaehehehe — CHANGES_REQUESTED** [review]

**Review by kaehehehe — COMMENTED** [review]
一度、Approveしてしまったんですが、修正をお願いします。申し訳ないです！

**Comment by kaehehehe** [comment]
Please revert all code related to 'id' and use 'useId' instead. Also, fix the code as follows.



```tsx
              <PopoverTrigger>
                <ColorPickerField
                  role="combobox" // ✅
                  aria-controls={id} // ✅
                  h={h}
                  minH={minH}
                  {...getFieldProps(inputProps, ref)}
                />
              </PopoverTrigger>

              {eyeDropperSupported && withEyeDropper ? (
                <ColorPickerEyeDropper
                  {...getEyeDropperProps(eyeDropperProps)}
                />
              ) : null}
            </ui.div>

            <Portal {...portalProps}>
              <PopoverContent
                containerProps={{ // ✅
                  id, 
                }}
                className="ui-color-picker__content"
                __css={{ ...styles.content }}
              >
                <ColorSelector
                  className="ui-color-picker__picker"
                  {...getSelectorProps({
                    saturationSliderRef,
                    saturationSliderProps,
                    swatchesProps,
                    hueSliderRef,
                    hueSliderProps,
                    alphaSliderRef,
                    alphaSliderProps,
                    channelsProps,
                    channelProps,
                  })}
                >
```

File: packages/components/color-picker/src/color-picker.tsx [hunk 3]
```diff
@@ -185,7 +187,13 @@ export const ColorPicker = forwardRef<ColorPickerProps, "input">(
 
             <Portal {...portalProps}>
               <PopoverContent
+                containerProps={{
+                  id,
+                }}
                 className="ui-color-picker__content"
+                role="combobox"
+                aria-controls={id}
+                aria-labelledby="color-picker-label"
```
Comment (kaehehehe) [inline packages/components/color-picker/src/color-picker.tsx:196]:
```suggestion
```

File: packages/components/color-picker/src/color-picker.tsx [hunk 4]
```diff
@@ -167,11 +168,12 @@ export const ColorPicker = forwardRef<ColorPickerProps, "input">(
               }}
             >
               {withSwatch ? <ColorPickerSwatch {...swatchProps} /> : null}
-
               <PopoverTrigger>
                 <ColorPickerField
                   h={h}
                   minH={minH}
+                  aria-expanded="false"
+                  aria-controls={id}
```
Comment (kaehehehe) [inline packages/components/color-picker/src/color-picker.tsx:176]: Please fix as below.
```suggestion
                  role="combobox"
                  aria-controls={id}
```

**Review by kaehehehe — CHANGES_REQUESTED** [review]

File: packages/components/color-picker/src/use-color-picker.ts [hunk 5]
```diff
@@ -400,6 +400,7 @@ export const useColorPicker = (props: UseColorPickerProps) => {
       ...containerProps,
       ...props,
       ...formControlProps,
+
```
Comment (kaehehehe) [inline packages/components/color-picker/src/use-color-picker.ts:403]: Please remove unnecessary line breaks.

File: packages/components/color-picker/src/use-color-picker.ts [hunk 6]
```diff
@@ -419,6 +420,7 @@ export const useColorPicker = (props: UseColorPickerProps) => {
       value: inputValue,
       "data-active": dataAttr(isOpen),
       "aria-expanded": dataAttr(isOpen),
+
```
Comment (kaehehehe) [inline packages/components/color-picker/src/use-color-picker.ts:423]: Please remove unnecessary line breaks.

**Review by kaehehehe — CHANGES_REQUESTED** [review]

**Review by kaehehehe — DISMISSED** [review]
Thanks!! :)

**Comment by hirotomoyamada** [comment]
Please resolve conflicts.

**Comment by hirotomoyamada** [comment]
@taku10101 

Please resolve conflicts.

**Review by kaehehehe — CHANGES_REQUESTED** [review]
Please check for build, ESLint, and TypeScript errors. @taku10101 
스크린샷 2024-10-07 20 50 48
스크린샷 2024-10-07 20 50 32
스크린샷 2024-10-07 20 50 17

**Review by kaehehehe — APPROVED** [review]
Thanks! 👍🏼 @taku10101

**Comment by hirotomoyamada** [comment]
There are too many commits for the PR. 

This PR is closed. Please put together your commitments again and submit them.

**Pull request closed without merging**